lib/enigma.rb

```diff
@@ -14,11 +14,67 @@
 # > e.crack(output) # Date is optional, use today's date
 # => "this is so secret ..end.."
 #
-# def encrypt
-# end
-#
-# def decrypt
-# end
-#
-# def crack
-# end
+require_relative 'key_generator'
+require_relative 'offset_calculator'
+
+
+class Enigma
+  attr_accessor :key, :oc
+
+  def initialize(oc = OffsetCalculator.new)
+    @counter = 0
+    @characters = ("a".."z").to_a
+    @oc = oc
+  end
+
+  def hash_zip(rotation = oc.offset)
+    rotated_characters = @characters.rotate(rotation)
```

Comment:
`rotate` cool! Ruby has everything.

Reply:
I might make rotated_characters it's own method:
```ruby
def rotated_characters(rotation = oc.offset)
  @characters.rotate(rotation)
end
```

Do you need to pass in rotation, or is it always going to be `oc.offest`?